main.go

```diff
@@ -732,12 +706,29 @@ func main() {
 			}
 		}()
 	}
+
 	settings, err := models.GetSettings(sqlDB)
 	if err != nil {
 		l.Fatalf("Failed to get settings: %+v.", err)
 	}
 
-	dbaasControllerClient := getDBaaSControllerClient(ctx, *dbaasControllerAPIAddrF, settings)
+	dbaasClient := dbaas.NewClient(*dbaasControllerAPIAddrF)
+	server.SetDbaasClient(dbaasClient)
+	if settings.DBaaS.Enabled {
+		l.Debug("DBaaS is enabled - creating a DBaaS client.")
+		ctx, cancel := context.WithTimeout(ctx, time.Second*20)
+		err := dbaasClient.Connect(ctx)
+		cancel()
+		if err != nil {
+			l.Fatalf("Failed to connect to dbaas-controller API on %s: %v", *dbaasControllerAPIAddrF, err)
+		}
+		defer func() {
+			err := dbaasClient.Disconnect()
+			if err != nil {
+				l.Fatalf("Failed to disconnect from dbaas-controller API: %v", err)
+			}
+		}()
+	}
```

Comment:
we can move all this block above https://github.com/percona/pmm-managed/pull/724/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R632

Reply:
No, we can't this needs to be after setup is called.

Reply:
Why?

Reply:
ok, I see. then we can just move https://github.com/percona/pmm-managed/pull/724/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R715 above and keep other lines here.

Reply:
Ok, good point.